ts/db_test.go

```diff
@@ -0,0 +1,351 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Matt Tracy (matt.r.tracy@gmail.com)
+
+package ts
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/gossip"
+	"github.com/cockroachdb/cockroach/kv"
+	"github.com/cockroachdb/cockroach/multiraft"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/rpc"
+	"github.com/cockroachdb/cockroach/storage"
+	"github.com/cockroachdb/cockroach/storage/engine"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/hlc"
+	gogoproto "github.com/gogo/protobuf/proto"
+)
+
+// TODO: These values were copied from storage/range_test.go in the
+// name of expediency. A better home should be found for common test defaults
+// like this.
+var (
+	testDefaultAcctConfig = proto.AcctConfig{}
+	testDefaultPermConfig = proto.PermConfig{
+		Read:  []string{"root"},
+		Write: []string{"root"},
+	}
+	testDefaultZoneConfig = proto.ZoneConfig{
+		ReplicaAttrs: []proto.Attributes{
+			{Attrs: []string{"dc1", "mem"}},
+			{Attrs: []string{"dc2", "mem"}},
+		},
+		RangeMinBytes: 1 << 10, // 1k
+		RangeMaxBytes: 1 << 18, // 256k
+		GC: &proto.GCPolicy{
+			TTLSeconds: 24 * 60 * 60, // 1 day
+		},
+	}
+)
+
+// tsTestModel is a model-based testing structure used to verify that time
+// series data sent to the Cockroach time series DB is stored correctly.
+//
+// This structure maintains a single ts.DB instance which stores data in a
+// monolithic Cockroach Store. It additionally maintains a simple in-memory key
+// value map, which is used as a model of the time series data stored in
+// Cockroach. The model maintains an expected copy of all keys beginning with
+// the time series data prefix.
+//
+// Each test should send a series of commands to the tsTestModel. Commands are
+// dispatched to the ts.DB instance, but are also used to modify the
+// in-memory key value model. Tests should periodically compare the in-memory
+// model to the actual data stored in the cockroach engine, ensuring that the
+// data matches.
+type tsTestModel struct {
```

Comment:
You're already in the `ts` package, no need to prefix stuff with `ts`: `s/tsTestModel/testModel/g`.